Sources/TuistKit/Generator/ProjectDescriptionHelpersBuilder.swift

```diff
@@ -33,42 +10,84 @@ protocol ProjectDescriptionHelpersBuilding: AnyObject {
     /// - Parameters:
     ///   - at: Path to the directory that contains the manifest being loaded.
     ///   - projectDescriptionPath: Path to the project description module.
-    func build(at: AbsolutePath, projectDescriptionPath: AbsolutePath) throws -> ProjectDescriptionHelpersModule?
+    func build(at: AbsolutePath, projectDescriptionPath: AbsolutePath) throws -> AbsolutePath?
 }
 
 final class ProjectDescriptionHelpersBuilder: ProjectDescriptionHelpersBuilding {
     /// Instance to locate the root directory of the project.
     let rootDirectoryLocator: RootDirectoryLocating
 
+    /// Path to the cache directory.
+    let cacheDirectory: AbsolutePath
+
     /// Initializes the builder with its attributes.
-    /// - Parameter rootDirectoryLocator: Instance to locate the root directory of the project.
-    init(rootDirectoryLocator: RootDirectoryLocating = RootDirectoryLocator()) {
+    /// - Parameters:
+    ///   - rootDirectoryLocator: Instance to locate the root directory of the project.
+    ///   - cacheDirectory: Path to the cache directory.
+    init(rootDirectoryLocator: RootDirectoryLocating = RootDirectoryLocator(),
+         cacheDirectory: AbsolutePath = Environment.shared.projectDescriptionHelpersCacheDirectory) {
         self.rootDirectoryLocator = rootDirectoryLocator
+        self.cacheDirectory = cacheDirectory
     }
 
-    func build(at: AbsolutePath, projectDescriptionPath: AbsolutePath) throws -> ProjectDescriptionHelpersModule? {
-        // We return if the helpers directory doesn't exist at /Tuist/ProjectDesciptionHelpers
+    func build(at: AbsolutePath, projectDescriptionPath: AbsolutePath) throws -> AbsolutePath? {
+        guard let helpersDirectory = self.helpersDirectory(at: at) else { return nil }
+        let hash = try self.hash(helpersDirectory: helpersDirectory)
+        let prefixHash = self.prefixHash(helpersDirectory: helpersDirectory)
+
+        // Get paths
+        let helpersCachePath = cacheDirectory.appending(component: prefixHash)
+        let helpersModuleCachePath = helpersCachePath.appending(component: hash)
+        let dylibName = "libProjectDescriptionHelpers.dylib"
+
+        if FileHandler.shared.exists(helpersModuleCachePath) {
+            return helpersModuleCachePath.appending(component: dylibName)
```

Comment:
As a further optimization, you could cache the known built dylib paths in an in memory cache (saves calculating hashes repeatedly for every manifest)